tools/topology/topology2/doc/topology2-generate-contents.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+# This script generates the contents page and stores it in extra-contents/contents.doxy
+
+generate_contents ()
+{
+	echo "# This contents list is automatically generated by $1."
+	echo "/*! \page doc_contents"
+	echo "    \section doc_classes Classes"
+
+	for ctype in Base Widget Control Dai Pipeline
+	do
+		echo "    \subsection doc_${ctype,}_classes $ctype Classes"
+		find .. -name \*.conf -print0 |
+		xargs -0 awk "/^Class\.$ctype\./ "'{
+			split($1, a, "\\.");
+			name = a[3]; gsub("\"", "", name); gsub("-", "_", name);
+			print "\t\\li \\ref " name
+		}' | sort -u
+		echo
+	done
+
+	echo '*/'
+}
+
+set -e
+cd "$(dirname "$0")"
+generate_contents "$0" > $1
```

Comment:
Move the `>` redirection to the `add_custom_command()` that invokes this. You just need to make sure there's white space on both sides of `>`